src/OmniSharp.Cake/Configuration/CakeConfiguration.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.IO;
+using OmniSharp.Cake.Configuration.Parser;
+
+namespace OmniSharp.Cake.Configuration
+{
+    [Export(typeof(ICakeConfiguration)), Shared]
+    internal sealed class CakeConfiguration : ICakeConfiguration
+    {
+        private readonly Dictionary<string, string> _lookup;
+
+        [ImportingConstructor]
+        public CakeConfiguration(IOmniSharpEnvironment environment)
+        {
+            _lookup = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            // Parse the configuration file.
+            var configurationPath = Path.Combine(environment.TargetDirectory, "cake.config");
+            if (File.Exists(configurationPath))
+            {
+                var parser = new ConfigurationParser();
```

Comment:
Parser will throw if the config is found among other things, if this throws in the constructor that may be a very very bad thing when booting up MEF. 

The parser should probably either not throw, or should we should catch specific exceptions that don't matter.